src/Packagist/WebBundle/Controller/ApiController.php

```diff
@@ -57,6 +57,41 @@ public function packagesAction(Request $req)
     }
 
     /**
+     * @Route("/api/create-package", name="generic_create", defaults={"_format" = "json"})
+     * @Method({"POST"})
+     */
+    public function createPackageAction(Request $request)
+    {
+
+        $payload = json_decode($request->getContent(), true);
+        if (!$payload) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Missing payload parameter'), 406);
+        }
+        $url = $payload['repository']['url'];
+        $package = new Package;
+        $package->setEntityRepository($this->getDoctrine()->getRepository('PackagistWebBundle:Package'));
+        $package->setRouter($this->get('router'));
+        $user = $this->findUser($request);
+        $package->addMaintainer($user);
+        $package->repository = $url;
+        $errors = $this->get('validator')->validate($package)
+        if (count($errors) > 0) {
+            $errorsString = (string) $errors;
```

Comment:
casting as string is a bad idea. The symfony doc says it can be handy for debugging. But it is clearly not handy for an API (you have no way to do something with it in your API client, except displaying it as is, and it is not very user-friendly).
You should rather build an array of errors in a format exposing useful info